src/main/java/net/tiagofar78/prisonescape/game/PrisonEscapeGame.java

```diff
@@ -578,6 +578,25 @@ public void run() {
         }, TICKS_PER_SECOND * _settings.getSecondsInSolitary());
     }
 
+    private void playerCompleteMission(PrisonEscapePlayer player) {
+        int missionPay = 1; // TODO
+        player.increaseBalance(missionPay);
+        // Send message
+    }
+
+    private void playerTriesToBuyItem(PrisonEscapePlayer player) {
+        int itemPrice = 1;
+
+        if (itemPrice > player.getBalance()) {
+            // Send message cannot buy
+            return;
+        }
+
+        player.decreaseBalance(itemPrice);
+        // Add item to inventory
```

Comment:
If you want you can already have a PrisonEscapeItem as argument and give it to player.

Reply:
good point